core/io/image.cpp

```diff
@@ -2734,6 +2735,24 @@ Error Image::compress(CompressMode p_mode, CompressSource p_source, ASTCFormat p
 Error Image::compress_from_channels(CompressMode p_mode, UsedChannels p_channels, ASTCFormat p_astc_format) {
 	ERR_FAIL_COND_V(data.is_empty(), ERR_INVALID_DATA);
 
+	// RenderingDevice only.
+	if (GLOBAL_GET("rendering/textures/vram_compression/compress_with_gpu")) {
+		switch (p_mode) {
+			case COMPRESS_BPTC: {
+				// BC7 is unsupported currently.
+				if ((format >= FORMAT_RF && format <= FORMAT_RGBE9995) && _image_compress_bptc_rd_func) {
+					_image_compress_bptc_rd_func(this, p_channels);
```

Comment:
I would return an error here so that if compression fails for whatever reason, this just continues down and ends up calling _image_compress_bptc_func

i.e.:
```
Error err = _image_compress_bptc_rd_func(this, p_channels);
if (err == OK) {
	return OK;
}
```